### PR TITLE
add check to avoid deploying to Prod AAT-only products

### PR DESCRIPTION
**Pull request by iguannago**

### CODE REVIEW

**Review by timja — CHANGES_REQUESTED** [review]
Can't you re-use the environment approvals code, that loads this file:
https://github.com/hmcts/cnp-jenkins-config/blob/master/environment-approvals.yml

and only deploy to prod if it's in that file instead?

File: vars/withCamundaOnlyPipeline.groovy [hunk 1]
```diff
@@ -33,6 +33,8 @@ def call(type, String product, String component, String s2sServiceName, String t
     angular: new AngularPipelineType(this, deploymentProduct, component)
   ]
 
+  def deployToProdFlag = ['wa'].contains(product)?false: true;
```
Comment (hannah38) [inline vars/withCamundaOnlyPipeline.groovy:36]: Hi, can you look at this line again. Looks overly complicated - would `!product.equals('wa')` do

**Comment by iguannago** [comment]
@timja , regarding your comment, yes, that's was my intention in the first place. It seems that the camunda deployment process bypasses that environment approvals check.

File: vars/withCamundaOnlyPipeline.groovy [hunk 2]
```diff
@@ -115,9 +115,13 @@ def call(type, String product, String component, String s2sServiceName, String t
       onMaster {
         def nonProdEnv = new Environment(env).nonProdName
         def prodEnv = new Environment(env).prodName
+        
         camundaPublish(s2sServiceName, nonProdEnv, product, tenantId)
-        camundaPublish(s2sServiceName, prodEnv, product, tenantId)
-
+        
+        approvedEnvironmentRepository(environment, metricsPublisher) {
```
Comment (timja) [inline vars/withCamundaOnlyPipeline.groovy:121]: you should be passing a string to this not a class, https://github.com/hmcts/cnp-jenkins-library/blob/task/deploy-to-prod-flag/vars/approvedEnvironmentRepository.groovy#L13

File: vars/withCamundaOnlyPipeline.groovy (hunk 1 above)
Comment (iguannago) [reply vars/withCamundaOnlyPipeline.groovy]: done

**Review by timja — APPROVED** [review]
Looks good, can you test this in your project please before we merge?

Change your `@Library('Infrastructure')` to `@Library('Infrastructure@task/deploy-to-prod-flag')`

**Pull request merged**